account_import.go

```diff
@@ -0,0 +1,155 @@
+		InfoMsg     bool
+	}{
+		UserPage:    p,
+		Collections: c,
+		Flashes:     []template.HTML{},
+	}
+
+	flashes, _ := getSessionFlashes(app, w, r, nil)
+	for _, flash := range flashes {
+		if strings.HasPrefix(flash, "SUCCESS: ") {
+			d.Message = strings.TrimPrefix(flash, "SUCCESS: ")
+		} else if strings.HasPrefix(flash, "INFO: ") {
+			d.Message = strings.TrimPrefix(flash, "INFO: ")
+			d.InfoMsg = true
+		} else {
+			d.Flashes = append(d.Flashes, template.HTML(flash))
+		}
+	}
+
+	showUserPage(w, "import", d)
+	return nil
+}
+
+func handleImport(app *App, u *User, w http.ResponseWriter, r *http.Request) error {
+	// limit 10MB per submission
+	r.ParseMultipartForm(10 << 20)
+	files := r.MultipartForm.File["files"]
+	var fileErrs []error
+	filesSubmitted := len(files)
+	var filesImported int
+	for _, formFile := range files {
+		file, err := formFile.Open()
+		if err != nil {
+			fileErrs = append(fileErrs, fmt.Errorf("failed to open form file: %s", formFile.Filename))
+			log.Error("import textfile: open from form: %v", err)
+			continue
+		}
+		defer file.Close()
+
+		tempFile, err := ioutil.TempFile("", "post-upload-*.txt")
+		if err != nil {
+			fileErrs = append(fileErrs, fmt.Errorf("failed to create temporary file for: %s", formFile.Filename))
+			log.Error("import textfile: create temp file: %v", err)
+			continue
+		}
+		defer tempFile.Close()
+
+		_, err = io.Copy(tempFile, file)
+		if err != nil {
+			fileErrs = append(fileErrs, fmt.Errorf("failed to copy file into temporary location: %s", formFile.Filename))
+			log.Error("import textfile: copy to temp: %v", err)
+			continue
+		}
+
+		info, err := tempFile.Stat()
+		if err != nil {
+			fileErrs = append(fileErrs, fmt.Errorf("failed to get file info of: %s", formFile.Filename))
+			log.Error("import textfile: stat temp file: %v", err)
+			continue
+		}
+		post, err := wfimport.FromFile(filepath.Join(os.TempDir(), info.Name()))
+		if err == wfimport.ErrEmptyFile {
+			// not a real error so don't log
+			_ = addSessionFlash(app, w, r, fmt.Sprintf("%s was empty, import skipped", formFile.Filename), nil)
+			continue
+		} else if err != nil {
+			fileErrs = append(fileErrs, fmt.Errorf("failed to read copy of %s", formFile.Filename))
+			log.Error("import textfile: file to post: %v", err)
+			continue
+		}
+
+		post.Collection = r.PostFormValue("collection")
+		coll, _ := app.db.GetCollection(post.Collection)
+		if coll == nil {
+			coll = &Collection{
+				ID: 0,
+			}
+		}
+
+		submittedPost := SubmittedPost{
+			Title:   &post.Title,
+			Content: &post.Content,
+			Font:    "norm",
+		}
+		if coll.ID != 0 && app.cfg.App.Federation {
+			token, err := app.db.GetAccessToken(u.ID)
+			if err != nil {
+				fileErrs = append(fileErrs, fmt.Errorf("failed to authenticate uploading: %s", formFile.Filename))
+				log.Error("import textfile: get accesstoken: %+v", err)
+				continue
+			}
+			ownedPost, err := app.db.CreateOwnedPost(&submittedPost, token, coll.Alias, app.cfg.App.Host)
```

Comment:
I think we should be calling `CreatePost()` here, too. I know the whole setup is a bit messy, but `CreateOwnedPost()` was meant for direct API usage, and if we use it here, we're going to fill up the database with access tokens that are only used once (and waste resources generating those tokens).

If this route was taken in order to get a `PublicPost` back, we should just duplicate the `PublicPost` creation logic from `CreateOwnedPost()` instead of going through token generation.